### PR TITLE
RedisStorage: add `delete` and `list_objects` methods, extend `store` method to support extra aioredis kwargs

### PR DESCRIPTION
**Pull request by pcanto-hopeit**

### CODE REVIEW

File: plugins/storage/redis/src/hopeit/redis_storage/__init__.py [hunk 1]
```diff
@@ -50,13 +50,34 @@ async def get(self, key: str, *, datatype: Type[DataObject]) -> Optional[DataObj
...
         return None
 
-    async def store(self, key: str, value: DataObject):
+    async def store(self, key: str, value: DataObject, **kwargs):
         """
         Stores value under specified key
 
         :param key: str
         :param value: DataObject, instance of dataclass annotated with @dataobject
+        :param **kwargs: extra args
```
Comment (leosmerling-hopeit) [inline plugins/storage/redis/src/hopeit/redis_storage/__init__.py:59]: clarify how kwargs will be used

File: plugins/storage/redis/src/hopeit/redis_storage/__init__.py [hunk 2]
```diff
@@ -50,13 +50,34 @@ async def get(self, key: str, *, datatype: Type[DataObject]) -> Optional[DataObj
...
+
         """
         assert self._conn
         payload_str = str(Payload.to_json(value))
-        await self._conn.set(key, payload_str)
+        await self._conn.set(key, payload_str, **kwargs)
```
Comment (leosmerling-hopeit) [inline plugins/storage/redis/src/hopeit/redis_storage/__init__.py:64]: Tests should check that kwargs are passed

File: plugins/storage/redis/src/hopeit/redis_storage/__init__.py [hunk 3]
```diff
@@ -50,13 +50,34 @@ async def get(self, key: str, *, datatype: Type[DataObject]) -> Optional[DataObj
...
+
+    async def delete(self, key: str):
```
Comment (leosmerling-hopeit) [inline plugins/storage/redis/src/hopeit/redis_storage/__init__.py:66]: 1) Test is missing for this method
2) this new method should be added to fs_storage as well

File: plugins/storage/redis/src/hopeit/redis_storage/__init__.py [hunk 4]
```diff
@@ -50,13 +50,34 @@ async def get(self, key: str, *, datatype: Type[DataObject]) -> Optional[DataObj
...
+        """
+        Delete specified key
+
+        :param key: str, key to be deleted
+        """
+        assert self._conn
+        await self._conn.delete(key)
+
+    async def list(self, pattern: str = "*")-> List[str]:
```
Comment (leosmerling-hopeit) [inline plugins/storage/redis/src/hopeit/redis_storage/__init__.py:75]: 1) Test is missing for this method
2) I am sure, but please check if signature is the same as in fs_storage

**Review by leosmerling-hopeit — CHANGES_REQUESTED** [review]

File: plugins/storage/redis/src/hopeit/redis_storage/__init__.py (hunk 3 above)
Comment (pcanto-hopeit) [reply plugins/storage/redis/src/hopeit/redis_storage/__init__.py]: added

File: plugins/storage/redis/src/hopeit/redis_storage/__init__.py (hunk 4 above)
Comment (pcanto-hopeit) [reply plugins/storage/redis/src/hopeit/redis_storage/__init__.py]: renamed to list_objects

File: plugins/storage/fs/test/unit/test_fs.py [hunk 5]
```diff
@@ -37,6 +38,7 @@ async def save_and_load_file():
     path = await fs.store(key, test_fs)
     assert path == f'/tmp/{key}/{key}.json'
     loaded = await fs.get(key, datatype=FsMockData)
+    await fs.delete(key)
```
Comment (leosmerling-hopeit) [inline plugins/storage/fs/test/unit/test_fs.py:41]: This does nothing? It should be checked at least if the remove method in the filesystem is called.

I would add a test with multiple keys to delete to check the for loops is working.

**Review by leosmerling-hopeit — APPROVED** [review]

**Pull request merged**